### PR TITLE
feat(gripper): add support for DVT encoder

### PR DESCRIPTION
**Pull request by ahiuchingau**
This PR adds z-axis encoder support for DVT gripper (requires #577).

### CODE REVIEW

File: gripper/firmware/interfaces_z_motor.cpp [hunk 1]
```diff
@@ -71,8 +71,13 @@ struct motion_controller::HardwareConfig motor_pins {
 /**
  * The motor hardware interface.
  */
+#if PCBA_PRIMARY_REVISION == 'c'
+static constexpr void* enc_handle = &htim7;
```
Comment (fsinapi) [inline gripper/firmware/interfaces_z_motor.cpp:75]: Should this be `htim8`?

**Review by fsinapi — COMMENTED** [review]
In addition to switching the TIM handle, you're going to have to edit some initialization parameters in `gripper/firmware/interfaces_z_motor.cpp` (and the simulator equivalent) to add parameters for the encoder.

File: gripper/firmware/interfaces_z_motor.cpp (hunk 1 above)
Comment (ahiuchingau) [reply gripper/firmware/interfaces_z_motor.cpp:75]: oh man yeah

File: gripper/firmware/utility_gpio.c [hunk 2]
```diff
@@ -3,6 +3,18 @@
 #include "platform_specific_hal_conf.h"
 #include "stm32g4xx_hal_gpio.h"
 
+#if PCBA_PRIMARY_REVISION == 'c'
```
Comment (sfoster1) [inline gripper/firmware/utility_gpio.c:6]: this feels like it should be a different PR since it's doing something pretty unrelated

**Review by sfoster1 — CHANGES_REQUESTED** [review]
Couple inline things, but also IMO when we add an `#ifdef` for a new revision, we should always make it open-ended for the future. If we add a bunch of `#if PCBA_PRIMARY_REVISION == 'c'`, then whenever we add new revision `d` we have to go and edit those to make them `#if PCBA_PRIMARY_REVISION == 'c' || PCBA_PRIMARY_REVISION == 'd'`, and if we forget we won't get compiler errors or anything we'll just silently get the old behavior.

Instead, we should start out by making it `#if PCBA_PRIMARY_REVISION != 'b' && PCBA_PRIMARY_REVISION != 'a'` (or, well, at some point we should add a `REVISION_GREATER_EQUAL` macro) so that whenever we add revision `d` we'll automatically get the `c` behavior to build on.

File: gripper/firmware/interfaces_z_motor.cpp [hunk 3]
```diff
@@ -116,16 +122,33 @@ static freertos_message_queue::FreeRTOSMessageQueue<
     can::messages::UpdateMotorPositionEstimationRequest>
     update_position_queue("Position Queue");
 
+#if PCBA_PRIMARY_REVISION == 'c'
```
[the remark this answers is not in the document]
Comment (ahiuchingau) [reply gripper/firmware/interfaces_z_motor.cpp:125]: This one can be collapsed to the block above, but the stallcheck stuff in line 138 will needs to be separate

File: gripper/firmware/utility_gpio.c (hunk 2 above)
Comment (ahiuchingau) [reply gripper/firmware/utility_gpio.c:6]: Oh I should've put it in the description; we had to switch these GPIOs out in order to add the new encoder. So I'm gonna try to keep these here, otherwise, the encoder wouldn't work properly.

File: gripper/firmware/interfaces_z_motor.cpp [hunk 4]
```diff
@@ -116,16 +122,33 @@ static freertos_message_queue::FreeRTOSMessageQueue<
...
+static constexpr float encoder_pulses = 1024.0;
+#else
+static constexpr float encoder_pulses = 0.0;
```
[the remark this answers is not in the document]
Comment (ahiuchingau) [reply gripper/firmware/interfaces_z_motor.cpp:128]: Fixed by #583
